crates/conf/macros/src/lib.rs

```diff
@@ -0,0 +1,261 @@
+
+                let inner_check_fns = &mut field_info.check_fns;
+
+                let mut check_fn;
+                for attr in &field.attrs {
+                    // get the check attribute with closures like this: #[check(|v| v > 0)]
+                    if attr.path.is_ident("check") {
+                        let closure = match attr.parse_args::<syn::ExprClosure>() {
+                            Ok(args) => args,
+                            Err(e) => {
+                                let mut error = syn::Error::new_spanned(
+                                    attr,
+                                    "expected a closure like: `#[check(|v| v > 0)]`",
+                                );
+                                error.combine(e);
+                                return error.to_compile_error().into();
+                            }
+                        };
+
+                        // The closure should take a reference to the field type and return a Result<(), String>
+                        if closure.inputs.len() != 1 {
+                            return syn::Error::new_spanned(
+                                closure.inputs,
+                                "expected a closure with one argument",
+                            )
+                            .to_compile_error()
+                            .into();
+                        }
+
+                        let arg = &closure.inputs[0];
+                        let arg = match arg {
+                            Pat::Type(arg) => arg,
+                            _ => {
+                                return syn::Error::new_spanned(
+                                    attr,
+                                    "expected a closure with one argument",
+                                )
+                                .to_compile_error()
+                                .into();
+                            }
+                        };
+                        // let arg_type = &arg.ty;
+                        let arg_type = quote! { &#field_type };
+
+                        let arg_name = &arg.pat;
+                        let arg_name = quote! { #arg_name };
+
+                        let body = &closure.body;
+                        check_fn = Some(quote! {
+                            (|#arg_name: #arg_type| {
+                                #body
+                            })
+                        });
+
+                        inner_check_fns.push(check_fn)
+                    } else if attr.path.is_ident("ensure") {
+                        // the ensure attribute has 2 inputs like this: #[ensure(move_margin > 0., "`move_margin` must be positive.")]
+                        // it uses the `ensure!()` macro from the anyhow crate
+
+                        // we need to get the two inputs separated by a comma separately. the first one is a boolean expression and the second one is a string literal
```

Comment:
Ditto, this line (and a few other lines) seems a bit two long. Note that rustfmt set comment width to 80 by default (https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#comment_width).